docs/src/main/sphinx/admin/properties-optimizer.rst

```diff
@@ -146,3 +146,14 @@ Specifies minimal bucket to task ratio that has to be matched or exceeded in ord
 to use table scan node partitioning. When the table bucket count is small
 compared to the number of workers, then the table scan is distributed across
 all workers for improved parallelism.
+
+``colocated-joins-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``false``
+* **Session property:** ``colocated_join``
+
+*(Experimental)* Enables colocated join execution. If enabled, when both sides
```

Comment:
few things:
1. that happens only when join is of repartitioned type. We don't colocate replicated joins
2. Data is fully loaded to memory. However, we avoid reshuffling the data over network thus saving CPU and network.

Reply:
Updated, please have a look.